cql-parser/src/main/scala/troy/cql/parser/CqlParser.scala

```diff
@@ -53,7 +55,25 @@ object CqlParser extends JavaTokenParsers
    */
   def constant: Parser[Constant] = {
     import Constants._
-    (string | uuid | number | boolean) ^^ Constant // | hex // TODO
+
+    def float = """[+-]?[0-9]*((\.[0-9]+([eE][+-]?[0-9]+)?[fF]?)|([fF])|([eE][+‌​-]?[0-9]+))\b""".r
+    def hex = "[0-9a-fA-F]".r
+    def uuid = s"$hex{8}-$hex{4}-$hex{4}-$hex{4}-$hex{12}".r
+    def blob = s"0(x|X)$hex+".r
+
+    def floatNum = float ^^ { s => new FloatConstant(s.toFloat) }
+    def nan = "NaN".r ^^^ Nan
```

Comment:
Can you add a test case that ensures "NAN", and "Nan" won't parse?
I can imagine myself "fixing" those to be case insensitive.